Doc/library/sqlite3.rst

```diff
@@ -2614,6 +2590,70 @@ With some adjustments, the above recipe can be adapted to use a
 instead of a :class:`~collections.namedtuple`.
 
 
+.. _sqlite3-howto-text-factory:
+
+How to create and use text factories
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+By default, :mod:`!sqlite3` adapts SQLite values with the ``TEXT`` data type
+using :class:`str`.
+This works well for correctly encoded UTF-8 text, but it will fail for invalid
+Unicode sequences and other encodings.
+To work around this, you can use a custom :attr:`~Connection.text_factory`.
+
+Because of SQLite's `flexible typing`_, it is not uncommon to encounter table
+columns with the ``TEXT`` data type, containing arbitrary data.
+To demonstrate, let's create a test database with an invalid Unicode sequence.
+We will use a `CAST expression`_ to coerce an invalid Unicode sequence,
+represented as a hexadecimal string ``X'619F'``,
+into the ``TEXT`` data type:
+
+.. testcode::
+
+   con = sqlite3.connect(":memory:")
+   con.executescript("""
+       CREATE TABLE test (data TEXT);
+       INSERT INTO test VALUES(CAST(X'619F' AS TEXT));
```

Comment:
Are you from Norway? Did you use some historical character encodings different from ISO-8859-1?

I think that you can use more real-world example here. Some Norwegian geographic or personal name containing non-ASCII characters encoded in historical Norwegian character encoding. It is better to use something different from Latin1 and cp1252 because they are used sometimes as default fallback encoding, and we want to show that it is not the case here.

Reply:
> Are you from Norway?

I am.

> Did you use some historical character encodings different from ISO-8859-1?

Not that I can recall. IIRC, most webpages back in the 90s used ISO-8859-1.

> I think that you can use more real-world example here.

I agree; real-world examples are always better than contrived examples.

> Some Norwegian geographic or personal name containing non-ASCII characters encoded in historical Norwegian character encoding. It is better to use something different from Latin1 and cp1252 because they are used sometimes as default fallback encoding, and we want to show that it is not the case here.

Let's try and find something :)

Reply:
Using a different iso-8859-X (where `X != 1`) might be a good compromise, even though all iso-8859-X encoding can decode any byte sequence without decoding errors (even though the result might be unexpected).

> encoded in historical Norwegian character encoding

I'm not aware of any specific encoding other than `iso-8859-1`.
The Runic Unicode block comes to mind, but afaik that can only be encoded by UTF-* encodings :upside_down_face:), and wouldn't solve the problem at hand.

Reply:
@encukou proposed to use a Czech example with iso-8859-2, and I think a Czech example is perfect here.

Reply:
I hope the example of 3b9f620aacac66c20344473a218af5d75959736f is sufficient for now. Also note that we can cook up a better example in a later PR.